internal/app/secretless/tcp_proxy_service.go

```diff
@@ -0,0 +1,147 @@
+package secretless
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"reflect"
+
+	validation "github.com/go-ozzo/ozzo-validation"
+
+	"github.com/cyberark/secretless-broker/internal"
+	"github.com/cyberark/secretless-broker/pkg/secretless/plugin/connector/tcp"
+)
+
+// Zeroizes the values of the fetched credentials. We don't want to
+// rely on garbage collection for this (it might be slow and/or only free them) so
+// we manually clear
+func zeroizeCredentials(backendCredentials map[string][]byte) {
+	for _, rawCredential := range reflect.ValueOf(backendCredentials).MapKeys() {
+		credential := rawCredential.String()
+		for byteIndex := range backendCredentials[credential] {
+			backendCredentials[credential][byteIndex] = 0
+		}
+	}
```

Comment:
This seems a bit of a roundabout way to do it I think. the map can return the key by itself without reflect and you don't need to turn that key into a string either.